src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -146,7 +175,29 @@ else if (isIntervalFile(arg)) {
     }
 
     /**
-     * Read a file of genome locations to process. The file may be in BED, Picard,
+     * Converts a Feature-containing file to a list of intervals
+     *
+     * @param parser GenomeLocParser for creating intervals
+     * @param featureFileName file containing Features to convert to intervals
+     * @return a List of intervals corresponding to the locations of the Features in the provided file
+     * @throws UserException.CouldNotReadInputFile if the provided file is not in a supported Feature file format
+     */
+    public static List<GenomeLoc> featureFileToIntervals( final GenomeLocParser parser, final String featureFileName ) {
+        final File featureFile = new File(featureFileName);
+        final FeatureCodec<? extends Feature, ?> codec = FeatureManager.getCodecForFile(new File(featureFileName));
+
+        try ( final FeatureDataSource<? extends Feature> dataSource = new FeatureDataSource<>(featureFile, codec) ) {
+            final List<GenomeLoc> featureIntervals = new ArrayList<>();
+
+            for ( final Feature feature : dataSource ) {
```

Comment:
Right, because using `Spliterators` everywhere even when you don't intend to split the input is sooo much clearer and more straightforward :)